File: truss-compiler/src/main/java/truss/compiler/support/Argument.java
package truss.compiler.support;

import org.apache.commons.lang.Validate;

public abstract class Argument {
    private final String option;
    private final String description;
    private final boolean mandatory;
    private boolean isProvided;

    protected Argument(String option, String description, boolean mandatory) {
        Validate.notNull(option, "option");
        Validate.notNull(description, "description");

        this.option = option;
        this.description = description;
        this.mandatory = mandatory;
    }

    public String getOption() {
        return option;
    }

    public String getDescription() {
        return description;
    }

    public boolean isProvided() {
        return isProvided;
    }

    public void setProvided(boolean isProvided) {
        this.isProvided = isProvided;
    }

    public boolean isMandatory() {
        return mandatory;
    }

    public abstract boolean allowMultiple();

    public abstract boolean hasParameter();

    public abstract void setValue(String value);
}

File: Truss.Compiler/Support/Argument.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;

namespace Truss.Compiler.Support {
    public abstract class Argument {
        protected Argument(string option, string description, bool mandatory) {
            if (option == null) {
                throw new ArgumentNullException("option");
            }
            if (description == null) {
                throw new ArgumentNullException("description");
            }

            Option = option;
            Description = description;
            IsMandatory = mandatory;
        }

        public string Option { get; private set; }

        public string Description { get; private set; }

        public bool IsProvided { get; set; }

        public bool IsMandatory { get; private set; }

        public abstract bool AllowMultiple { get; }

        public abstract bool HasParameter { get; }

        public abstract void SetValue(string value);
    }
}
